src/main/java/io/cdap/plugin/gcp/dataplex/source/DataplexBatchSource.java

```diff
@@ -0,0 +1,514 @@
+        String.format("Read from BigQuery Entity %s '%s' from Dataplex.",
+          type, table));
+    }
+  }
+
+  /**
+   * this method will set up all the configuration to run the job in case of Cloud Storage entities.
+   *
+   * @param context BatchSourceContext
+   * @throws InstantiationException
+   * @throws IOException
+   * @throws ExecutionException
+   * @throws InterruptedException
+   */
+  private void prepareRunStorageBucket(BatchSourceContext context)
+    throws InstantiationException, IOException, ExecutionException, InterruptedException {
+    FailureCollector collector = context.getFailureCollector();
+    Job job = JobUtils.createInstance();
+    configuration = job.getConfiguration();
+
+    // Get storage and create temporary bucket to store task execution data from.
+    Storage storage = GCPUtils.getStorage(config.getProject(), config.getCredentials());
+    createBucket(configuration, storage, config.getLocation(),
+      GCS_TEMP_BUCKET_NAME);
+    String outputLocation = GCSPath.SCHEME + GCS_TEMP_BUCKET_NAME;
+    String query = formatQuery(entity, context.isPreviewEnabled());
+
+    // Create Dataplex task to fetch filtered data. And set in configuration
+    String taskId = createTask(outputLocation, query);
+    setConfigurationForDataplex(taskId);
+
+    // Set Validating Input Format
+    ValidatingInputFormat validatingInputFormat = config.getValidatingInputFormat(context);
+    FileInputFormat.setInputDirRecursive(job, true);
+
+    Schema schema = DataplexUtil.getTableSchema(entity.getSchema(), collector);
+    LineageRecorder lineageRecorder = new LineageRecorder(context, config.getReferenceName());
+    lineageRecorder.createExternalDataset(schema);
+
+    if (schema != null && schema.getFields() != null) {
+      this.recordLineage(lineageRecorder, schema.getFields().stream().map(Schema.Field::getName).collect(
+        Collectors.toList()), "Read from GCS entity in Dataplex.");
+    }
+
+    Iterator propertiesIterator = config.getFileSystemProperties(outputLocation).entrySet().iterator();
+    while (propertiesIterator.hasNext()) {
+      Map.Entry<String, String> entry = (Map.Entry) propertiesIterator.next();
+      configuration.set(entry.getKey(), entry.getValue());
+    }
+    Path path = new Path(outputLocation);
+    FileSystem pathFileSystem = FileSystem.get(path.toUri(), configuration);
+    FileStatus[] fileStatus = pathFileSystem.globStatus(path);
+    if (fileStatus == null) {
+      throw new IOException(String.format("Input path %s does not exist", path));
+    } else {
+      FileInputFormat.addInputPath(job, path);
+      Map<String, String> inputFormatConfiguration = validatingInputFormat.getInputFormatConfiguration();
+      Iterator inputFormatIterator = inputFormatConfiguration.entrySet().iterator();
+      while (inputFormatIterator.hasNext()) {
+        Map.Entry<String, String> propertyEntry = (Map.Entry) inputFormatIterator.next();
```

Comment:
You should be able to remove the casting here and declare the Iterator as `Iterator<> inoutFormatIterator = ...`

Reply:
done